src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java

```diff
@@ -352,7 +355,35 @@ private boolean sendMail(ExtendedEmailPublisherContext context) {
 
                     while (true) {
                         try {
-                            Transport.send(msg);
+                            ExtendedEmailPublisherDescriptor descriptor = getDescriptor();
+                            String smtpHost = descriptor.getSmtpServer();
+                            if (smtpHost == null) {
+                                smtpHost = "";
+                            }
+                            String awsRegex = "^email-smtp\\.([a-z0-9-]+)\\.amazonaws\\.com$";
+                            Pattern p = Pattern.compile(awsRegex);
+                            Matcher m = p.matcher(smtpHost);
+                            if (m.matches()) {
+                                String region = m.group(1);
+                                Session session = descriptor.createSession();
+                                Transport transport = session.getTransport(allRecipients[0]);
+                                try {
+                                    transport.connect();
+                                    transport.sendMessage(msg, allRecipients);
+                                    if (transport instanceof SMTPTransport) {
+                                        String response = ((SMTPTransport)transport).getLastServerResponse();
+                                        String[] parts = response.trim().split(" +");
+                                        if (parts.length == 3 && parts[0].equals("250") && parts[1].equals("Ok")) {
+                                            String MessageID = "<" + parts[2] + "@" + region + ".amazonses.com>";
+                                            msg.setHeader("Message-ID", MessageID);
+                                        }
```

Comment:
Yeah, I think I could accept something like that.

On Wed, Oct 14, 2015 at 1:39 PM weisslj <notifications@github.com> wrote:

> In src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java
> <https://github.com/jenkinsci/email-ext-plugin/pull/108#discussion_r42049715>
> :
>
> > +                            Pattern p = Pattern.compile(awsRegex);
> > +                            Matcher m = p.matcher(smtpHost);
> > +                            if (m.matches()) {
> > +                                String region = m.group(1);
> > +                                Session session = descriptor.createSession();
> > +                                Transport transport = session.getTransport(allRecipients[0]);
> > +                                try {
> > +                                    transport.connect();
> > +                                    transport.sendMessage(msg, allRecipients);
> > +                                    if (transport instanceof SMTPTransport) {
> > +                                        String response = ((SMTPTransport)transport).getLastServerResponse();
> > +                                        String[] parts = response.trim().split(" +");
> > +                                        if (parts.length == 3 && parts[0].equals("250") && parts[1].equals("Ok")) {
> > +                                            String MessageID = "<" + parts[2] + "@" + region + ".amazonses.com>";
> > +                                            msg.setHeader("Message-ID", MessageID);
> > +                                        }
>
> Yes, suspected that it might be too specific to include it. Would you
> accept a patch for an additional "post-send" script where the result of the
> last SMTP answer is available?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/email-ext-plugin/pull/108/files#r42049715>.
>